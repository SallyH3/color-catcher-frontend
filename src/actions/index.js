export const makePalette = (palette) => {
	return {
		type: 'MAKE_PALETTE',
		palette
	}
}

export const clearPalette = (palette) => {
	return {
		type: 'CLEAR_PALETTE',
		palette
	}
}