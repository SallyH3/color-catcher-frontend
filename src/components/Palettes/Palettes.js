import React, { Component } from 'react';
// import './_Palettes.scss';

class Palettes extends Component {
	constructor() {
		super();
		this.state = {

		}
	}

	render() {
		return (
			<div>
				<h1>Palettes</h1>
			</div>)
	}
}

export default Palettes;