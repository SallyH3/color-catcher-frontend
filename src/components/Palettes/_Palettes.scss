.palette-holder {
	background-color: #F8EFEF;
	width: 100%;
	min-height: 200px;
	height: 100%;
}

.picked-color {
	width: 20px;
	height: 20px;
	margin: 5px;
}

.color-holder {
	display: flex;
	padding-top: 20px;
}

.palette-container {
	display: flex;
}
