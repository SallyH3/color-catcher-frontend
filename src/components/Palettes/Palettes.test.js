import React from 'react';
import { shallow } from 'enzyme';
import { Palettes } from './Palettes';
import { addPalette, deletePalette } from '../../actions';

describe('Palettes', () => {
	let wrapper;
	let mockProject;
	let mockPalette;
	let mockPalettes;
	let mockAddPalette;
	let mockDeletePalette;

	beforeEach(() => {
		mockAddPalette = jest.fn();
		mockDeletePalette = jest.fn();
		mockProject = {id: 29, name: "named Ryan"};
		mockPalette = [
			{	locked: false,
				color: "#816379"
			},
			{ locked: false,
				color: "#816379"
			},
			{ locked: false,
				color: "#816379"
			},
			{	locked: false,
				color: "#816379"
			},
			{ locked: false,
				color: "#816379"
			}
		];
		mockPalettes = [
		{	color_1: "CF71BD",
			color_2: "D09B24",
			color_3: "0C2A28",
			color_4: "CC3EDC",
			color_5: "816379",
			name: "hi there",
			project_id: 30,
			id: 40},
		{	color_1: "CF71BB",
			color_2: "D09B22",
			color_3: "002A28",
			color_4: "CC3EDC",
			color_5: "886379",
			name: "bye there",
			project_id: 31,
			id: 41}
		];


		wrapper = shallow(
			<Palettes 
				addPalette={mockAddPalette}
				deletePalette={mockDeletePalette}
				project={mockProject}
				palette={mockPalette}
				palettes={mockPalettes}
			/>
		)
	});

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('handleChange', () => {
		it('should handle the change of name', () => {
			const mockNameEvent = {
				target: {
					value: 'Palette One'
				}
			};

			wrapper.instance().handleChange(mockNameEvent);
			expect(wrapper.state().name).toEqual('Palette One')
		});
	});

	describe('addPalette', () => {
		it('should call post palette', () => {
			const newPalette = {	
				color_1: "CF71BD",
				color_2: "D09B24",
				color_3: "0C2A28",
				color_4: "CC3EDC",
				color_5: "816379",
				name: "hi there",
				project_id: 30,
				id: 40
			}
			wrapper.instance().postPalette = jest.fn();
			wrapper.instance().addPalette();
			expect(wrapper.instance().postPalette).toHaveBeenCalled();
		});
	});

	describe('postPalette', () => {
		it.skip('should have a post fetch', () => {

		});
	});

	describe('deletePalette', () => {
		it.skip('should call a delete fetch', () => {

		});
		it('should dispatch a delete palette action', () => {

			const paletteIdToDelete = 1;
			wrapper.instance().deletePalette = jest.fn();

			wrapper.instance().deletePalette = jest.fn()
			wrapper.instance().deletePalette()
			expect(wrapper.instance().deletePalette).toHaveBeenCalled()
		})
	})

});