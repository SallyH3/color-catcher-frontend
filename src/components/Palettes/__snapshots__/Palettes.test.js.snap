// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palettes should match the snapshot 1`] = `
<div
  className="palette-holder"
>
  <h1>
    project
  </h1>
  <div>
    <input
      className="palette-name"
      onChange={[Function]}
      placeholder="Name your palette!"
      type="text"
      value=""
    />
    <button
      className="add-palette-btn"
      onClick={[Function]}
    >
      Add Palette 
    </button>
  </div>
</div>
`;
